image/preseed/preseed_linux.go

```diff
@@ -331,9 +350,20 @@ func prepareCore20Mountpoints(opts *preseedCoreOptions) (cleanupMounts func(), e
 		}
 	}
 
-	underWritable := func(path string) string {
-		return filepath.Join(opts.WritableDir, path)
+	if err := os.MkdirAll(filepath.Dir(currentLink), 0755); err != nil {
+		return nil, err
+	}
+	if err := os.MkdirAll(currentSnapdMountPoint, 0755); err != nil {
+		return nil, err
 	}
+	errV := os.Rename(currentLink, currentLink+".preseeding-backup")
```

Comment:
I think in a clean chroot, 'current' shouldn't exist unless we're calling reset or retrying preseeding.

Reply:
Yes, I think that is true in `prepareCore20Mountpoints`. And this could be simplified. But in `prepareClassicChroot` I still need to do that. Do we agree?